packages/batch-submitter/src/batch-submitter/state-batch-submitter.ts

```diff
@@ -202,9 +202,10 @@ export class StateBatchSubmitter extends BatchSubmitter {
       [...Array(blockRange).keys()],
       async (i: number) => {
         this.log.debug('Fetching L2BatchElement', { blockNo: startBlock + i })
-        const block = (await this.l2Provider.getBlockWithTransactions(
-          startBlock + i
-        )) as L2Block
+        const block = (await this.l2Provider.send('eth_getBlockByNumber', [
+          toRpcHexString(startBlock + i),
+          true,
+        ])) as L2Block
```

Comment:
This particular change is due to migration away from the `OptimismProvider`?

Reply:
Yes that's what the code I linked to above does! The parts of the Optimism Provider made in there are the ones which matter here (injecting extra l2 context)